src/Workspaces/Core/Portable/Shared/Utilities/BloomFilter.cs

```diff
@@ -334,22 +335,19 @@ public void Add(long value)
     {
         for (var i = 0; i < _hashFunctionCount; i++)
         {
-            _bitArray[GetBitArrayIndex(value, i)] = true;
+            var hash = ComputeHash(value, i);
+            _bitArray[GetBitArrayIndexFromHash(hash)] = true;
         }
     }
 
-    private int GetBitArrayIndex(long value, int i)
-    {
-        var hash = ComputeHash(value, i);
-        hash %= _bitArray.Length;
-        return Math.Abs(hash);
-    }
-
     public bool ProbablyContains(string value)
     {
+        var hashes = BloomFilterHash.GetOrCreateHashArray(value, _isCaseSensitive, _hashFunctionCount);
```

Comment:
ok.  this is subtle, and needs docs.  mention explicitly that hashes may contain *more* hash values than the _hashFunctionCount passed in.  But that's ok as the first `_hashFunctionCount` hashes are guaranteed to be the same due to how ComputeHash works.